src/query_graph/mod.rs

```diff
@@ -723,9 +723,44 @@ impl QueryGraph {
 
     pub(crate) fn has_an_implementation_with_provides(
         &self,
-        _source: &NodeStr,
-        _interface_field_definition_position: InterfaceFieldDefinitionPosition,
+        source: &NodeStr,
+        interface_field_definition_position: InterfaceFieldDefinitionPosition,
     ) -> Result<bool, FederationError> {
-        todo!()
+        let schema = self.schema_by_source(source)?;
+        let Some(metadata) = schema.subgraph_metadata() else {
+            return Err(FederationError::internal(format!(
+                "Interface should have come from a federation subgraph {}",
+                source
+            )));
+        };
+
+        let provides_directive_definition = metadata
+            .federation_spec_definition()
+            .provides_directive_definition(schema)?;
+        let type_name = interface_field_definition_position.parent();
+        let interface_type = type_name.get(schema.schema())?;
+
+        if let Some(implements_interface) = interface_type.implements_interfaces.iter().next() {
```

Comment:
This looks the wrong way around. In JS we are checking all object types that implement this interface, but `.implements_interfaces` contains the super-interfaces of this interface.

The Rust codebase has a `schema.possible_runtime_types(position)?` method for accessing that information.

It should also be a loop so we check every implementation and not just the first

Reply:
ohhhh i didn't find this function! with that this looks a lot better now